src/main/java/com/openshift/internal/restclient/WatchClient.java

```diff
@@ -52,9 +58,68 @@
 	private IResourceFactory factory;
 	private IClient client;
 	private static WebSocketClient wsClient;
+	private static AtomicReference<Status> status = new AtomicReference<>(Status.Stopped);
+	private static Lock lock = new ReentrantLock();
+	private static Condition isStarted = lock.newCondition();
+	private static long lockTimeout; 
+	
+	private enum Status{
+		Started,
+		Starting,
+		Stopped,
+		Stopping
+	}
 	
 	static {
-		wsClient = newWebSocketClient();
+		try {
+			lockTimeout = Long.parseLong(System.getProperty("com.openshift.restclient.watchlocktimeoutms", "30000")); 
+		}catch(NumberFormatException e) {
+			lockTimeout = 30 * 1000;
+		}
```

Comment:
can we please extract this try/catch-block to a method with a speaking name like #getLockTimeout?